Packs/FeedSharedTenantElasticsearch/Integrations/FeedElasticsearch/FeedElasticsearch.py

```diff
@@ -0,0 +1,345 @@
+        if tenant_hash:
+            # all shared indexes minus this tenant shared
+            indices += f',-*{tenant_hash}*-shared*'
+    elif not indices:
+        indices = '_all'
+    search = Search(using=es, index=indices).filter({'range': range_field}).query(query)
+    return search
+
+
+def extract_indicators_from_insight_hit(hit):
+    """Extracts indicators from an insight hit including enrichments"""
+    ioc_lst = []
+    ioc_enirhcment_list = []
+    ioc = hit_to_indicator(hit)
+    if ioc.get('value'):
+        ioc_lst.append(ioc)
+        module_to_feedmap = ioc.get(MODULE_TO_FEEDMAP_KEY)
+        updated_module_to_feedmap = {}
+        if module_to_feedmap:
+            ioc_enrichment_obj = []
+            for key, val in module_to_feedmap.items():
+                if val.get('isEnrichment'):
+                    ioc_enrichment_obj.append(val)
+                else:
+                    updated_module_to_feedmap[key] = val
+            if ioc_enrichment_obj:
+                ioc_enirhcment_list.append(ioc_enrichment_obj)
+            ioc[MODULE_TO_FEEDMAP_KEY] = updated_module_to_feedmap
+    return ioc_lst, ioc_enirhcment_list
+
+
+def hit_to_indicator(hit, ioc_val_key='name', ioc_type_key=None, default_ioc_type=None):
+    """Convert a single hit to an indicator"""
+    ioc_dict = hit.to_dict()
+    ioc_dict['value'] = ioc_dict.get(ioc_val_key)
+    ioc_dict['rawJSON'] = dict(ioc_dict)
+    if default_ioc_type:
+        ioc_dict['type'] = default_ioc_type
+    elif ioc_type_key:
+        ioc_dict['type'] = ioc_dict.get(ioc_type_key)
+    return ioc_dict
+
+
+def create_enrichment_batches(ioc_enrch_lst):
+    """
+    Create batches for enrichments, by separating enrichments that come from the same indicator into diff batches
+    """
+    max_enrch_len = 0
+    for ioc_enrch_obj in ioc_enrch_lst:
+        max_enrch_len = max(max_enrch_len, len(ioc_enrch_obj))
+    enrch_batch_lst = []
+    for i in range(max_enrch_len):
+        enrch_batch_obj = []
+        for ioc_enrch_obj in ioc_enrch_lst:
+            if i < len(ioc_enrch_obj):
+                enrch_batch_obj.append(ioc_enrch_obj[i])
+        enrch_batch_lst.append(enrch_batch_obj)
+    return enrch_batch_lst
+
+
+def main():
+    try:
+        LOG('command is %s' % (demisto.command(),))
+        params = demisto.params()
+        server = params.get('url', '').rstrip('/')
+        creds = params.get('credentials')
+        username, password = (creds.get('identifier'), creds.get('password')) if creds else (None, None)
+        insecure = not params.get('insecure')
+        feed_type = params.get('feed_type')
+        time_field = params.get('time_field') if feed_type == FEED_TYPE_GENERIC else 'calculatedTime'
+        time_method = params.get('time_method')
+        fetch_index = params.get('fetch_index')
+        fetch_time = params.get('fetch_time', '3 days')
+        query = params.get('es_query')
+        api_key = params.get('api_key')
```

Comment:
same here about the API key vs the username pass

Reply:
see my previous comment